### DISCUSSION TITLE
Dynamic column accessor based on context

### DISCUSSION THREAD

**Discussion by efreila** [TanStack/table#3007] [Q&A]
I currently have a column in my table that is supposed to show one of two different values based on a boolean stored in a global context. Ideally, to achieve this, I would be able to useContext() in my accessor function and return an accessor value based on the context state. I currently pass both pieces of data into the useTable() hook, so it's available in each row.  However, since "accessor" is not a custom hook or component, I can't useContext() in it. I thought of a few other solutions but wasn't sure what the best approach would be.

1. Leave out accessor altogether and simply display the proper data by putting the logic in a custom "Cell"
2. Modify the columns array that is passed in to the useTable() hook every time the global context state is modified

Would love to discuss and hear your suggestions to get this working.

**Comment by yannxaver** [comment]
@efreila How did you solve it?
